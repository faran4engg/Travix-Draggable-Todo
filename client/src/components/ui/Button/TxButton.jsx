import React from "react";
import PropTypes from "prop-types";
import { Button, Icon } from "semantic-ui-react";

const TxButton = props => {
  const {
    type,
    positive,
    disabled,
    color,
    animated,
    floated,
    click,
    content,
    iconName,
    size,
    fluid
  } = props;
  return (
    <Button
      type={type}
      positive={positive}
      disabled={disabled}
      color={color}
      animated={animated && animated}
      floated={floated}
      onClick={click}
      size={size}
      fluid={fluid}
    >
      {animated ? (
        <React.Fragment>
          <Button.Content hidden>{content}</Button.Content>
          <Button.Content visible>
            <Icon name={iconName} />
          </Button.Content>
        </React.Fragment>
      ) : (
        <Button.Content>{content}</Button.Content>
      )}
    </Button>
  );
};

TxButton.defaultProps = {
  animated: false
};

TxButton.propTypes = {
  type: PropTypes.string,
  positive: PropTypes.bool,
  disabled: PropTypes.bool,
  color: PropTypes.string,
  animated: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  floated: PropTypes.string,
  click: PropTypes.func,
  content: PropTypes.string,
  iconName: PropTypes.string,
  size: PropTypes.string,
  fluid: PropTypes.bool
};

export default TxButton;
