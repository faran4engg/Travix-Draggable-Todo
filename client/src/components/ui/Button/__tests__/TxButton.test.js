import React from "react";

import { shallow, mount } from "enzyme";
import TxButton from "../TxButton";

describe("TxButton", () => {
  it("should render correctly ", () => {
    const component = shallow(<TxButton content="Delete" />);
    expect(component).toMatchSnapshot();
  });
  describe("when animated props is not passed", () => {
    it("should show button text ", () => {
      const component = mount(<TxButton content="Delete" />);
      expect(component.find("TxButton").text()).toEqual("Delete");
    });

    it("should not show animated icon ", () => {
      const component = shallow(<TxButton />);
      expect(component.find("Icon")).toHaveLength(0);
    });
  });
  describe("when animated props is passed", () => {
    it("should show button text ", () => {
      const component = mount(<TxButton animated="fade" content="Edit" />);
      expect(component.find("TxButton").text()).toEqual("Edit");
    });

    it("should show animated icon ", () => {
      const component = shallow(<TxButton animated="fade" />);
      expect(component.find("Icon")).toHaveLength(1);
    });
  });
});
