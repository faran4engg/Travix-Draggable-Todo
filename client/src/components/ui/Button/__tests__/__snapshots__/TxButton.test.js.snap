// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxButton should render correctly  1`] = `ShallowWrapper {}`;
